pkg/instance/helmfile.go

```diff
@@ -100,6 +129,55 @@ func (h helmfileService) loadStackParameters(stackName string) (stackParameters,
 	return params, nil
 }
 
+func configureInstanceEnvironment_deployment(accessToken string, stack *model.Stack, instance *model.DeploymentInstance, group *model.Group, stackParameters stackParameters, cmd *exec.Cmd) {
+	// TODO: We should only inject what the stack require, currently we just blindly inject IM_ACCESS_TOKEN and others which may not be required by the stack
+	// We could probably list the required system parameters in the stacks helmfile and parse those as well as other parameters
+	instanceNameEnv := fmt.Sprintf("%s=%s", "INSTANCE_NAME", instance.Name)
+	instanceNamespaceEnv := fmt.Sprintf("%s=%s", "INSTANCE_NAMESPACE", group.Name)
+	instanceIdEnv := fmt.Sprintf("%s=%d", "INSTANCE_ID", instance.ID)
+	// instanceTTLEnv := fmt.Sprintf("%s=%d", "INSTANCE_TTL", instance.TTL)
+	instanceTTLEnv := fmt.Sprintf("%s=%d", "INSTANCE_TTL", 3800)
+	instanceHostnameEnv := fmt.Sprintf("%s=%s", "INSTANCE_HOSTNAME", group.Hostname)
+	imTokenEnv := fmt.Sprintf("%s=%s", "IM_ACCESS_TOKEN", accessToken)
+	homeEnv := fmt.Sprintf("%s=%s", "HOME", "/tmp")
+	imCreationTimestamp := fmt.Sprintf("%s=%d", "INSTANCE_CREATION_TIMESTAMP", time.Now().Unix())
+	cmd.Env = append(cmd.Env, instanceNameEnv, instanceNamespaceEnv, instanceIdEnv, instanceTTLEnv, instanceHostnameEnv, imTokenEnv, homeEnv, imCreationTimestamp)
+
+	cmd.Env = injectEnv(cmd.Env, "HOSTNAME")
+	cmd.Env = injectEnv(cmd.Env, "AWS_ACCESS_KEY_ID")
+	cmd.Env = injectEnv(cmd.Env, "AWS_SECRET_ACCESS_KEY")
+	cmd.Env = injectEnv(cmd.Env, "AWS_DEFAULT_REGION")
+	cmd.Env = injectEnv(cmd.Env, "AWS_REGION")
+	cmd.Env = injectEnv(cmd.Env, "AWS_ROLE_ARN")
+	cmd.Env = injectEnv(cmd.Env, "AWS_WEB_IDENTITY_TOKEN_FILE")
+
+	cmd.Env = injectEnv(cmd.Env, "KUBERNETES_SERVICE_PORT")
+	cmd.Env = injectEnv(cmd.Env, "KUBERNETES_PORT")
+	cmd.Env = injectEnv(cmd.Env, "KUBERNETES_PORT_443_TCP_ADDR")
+	cmd.Env = injectEnv(cmd.Env, "KUBERNETES_PORT_443_TCP_PORT")
+	cmd.Env = injectEnv(cmd.Env, "KUBERNETES_PORT_443_TCP_PROTO")
+	cmd.Env = injectEnv(cmd.Env, "KUBERNETES_PORT_443_TCP")
+	cmd.Env = injectEnv(cmd.Env, "KUBERNETES_SERVICE_PORT_HTTPS")
+	cmd.Env = injectEnv(cmd.Env, "KUBERNETES_SERVICE_HOST")
+
+	for name, parameter := range stack.Parameters {
```

Comment:
Why are we not ranging over `instance.Parameters`? I thought in here the `DeploymentInstance` is all that is needed 🤔